### PR TITLE
Monobank_Api_Backend 

### PR DESCRIPTION
**Pull request by MementoMorj**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by ormykhalyshyn — APPROVED** [review]
Good clean code.
Nice job)

File: Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        public async Task<InvoiceInfo> CreateInvoiceAsync(Invoice invoice)
+        {
+            var (code, body) = await PostAsync(Api.Merchant.Invoice.Create, invoice);
+            
+            return code switch
+            {
+                200 => JsonConvert.DeserializeObject<InvoiceInfo>(body)
+            };
+        }
```
Comment (Katerix) [inline Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs:29]: i think we need to add 4XX-responses here

File: Streetcode/Streetcode.DAL/Entities/Payment/BasketOrder.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿using Newtonsoft.Json;
+
+namespace Streetcode.DAL.Entities.Payment
+{
+    // Root myDeserializedClass = JsonConvert.DeserializeObject<Root>(myJsonResponse);
+    public class BasketOrder
```
Comment (Katerix) [inline Streetcode/Streetcode.DAL/Entities/Payment/BasketOrder.cs:6]: remove this comment pls

**Review by Katerix — APPROVED** [review]
good job! unbelievable amount of work is done! i wonder who inspired you!

File: Streetcode/Streetcode.BLL/DTO/Payment/PaymentResponseDTO.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/DTO/Payment/PaymentResponseDTO.cs:5]: Maybe better ro delete this part of code?

File: Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    public class PaymentService : IPaymentService
+    {
+        private readonly HttpClient _httpClient;
+        
+        public PaymentService()
+        {
+            _httpClient = new HttpClient();
+            _httpClient.BaseAddress = new Uri(Api.Production);
+            _httpClient.DefaultRequestHeaders.Clear();
+            _httpClient.DefaultRequestHeaders.Add(RequestHeaders.XToken, "mo4jMO2iJmtwCB0RKw8CMBw");
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs:18]: This token could be injected inside code on the Github?

File: Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public async Task<InvoiceInfo> CreateInvoiceAsync(Invoice invoice)
+        {
+            var (code, body) = await PostAsync(Api.Merchant.Invoice.Create, invoice);
+            
+            return code switch
+            {
+                200 => JsonConvert.DeserializeObject<InvoiceInfo>(body)
+            };
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/Services/Payment/PaymentService.cs:28]: Switch pattern = +10 social rating.

File: Streetcode/Streetcode.DAL/Entities/Payment/Invoice.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        // [JsonProperty("validity")]
+        // public long Validity { get; set; }
+
+        // [JsonProperty("paymentType")]
+        // public string PaymentType { get; set; }
+
+        // [JsonProperty("qrId")]
+        // public string QrId { get; set; }
+
```
Comment (Kasterov) [inline Streetcode/Streetcode.DAL/Entities/Payment/Invoice.cs:38]: This comments should be here ot you will delete it here?

**Pull request merged**